src/main/java/com/redislabs/Connector.java

```diff
@@ -245,20 +235,26 @@ public HashMap<String, Object> map(HashMap<String, Object> r) throws Exception {
     return r;
   }
 
+
   @Override
   public void foreach(ArrayList<HashMap<String, Object>> record) throws Exception {
     String lastStreamId = null;
     String msg = null;
     Exception cause = null;
+    int lastCommittedIdx = -1;
     synchronized (this.connector) {
       try {
         Session session = connector.getSession();
         Transaction transaction = session.beginTransaction();
         boolean isMerge = true;
-        
+
+        int idx = -1;
         for(Map<String, Object> r: record) {
```

Comment:
Probably will be better to use a regular for here (`for(int idx = 0 ; idx < record.size(); ++idx)`). Or maybe there is some way to iterate it with an index?

Reply:
Made the change to use index